### DISCUSSION TITLE
Develop a software that can help me automate all sshkey-related processes

### DISCUSSION THREAD

**Discussion by vaynevayne** [community/community#55351] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

I have multiple github accounts and multiple gitlab accounts, and every time I change my account, it always leads to permission problems. This one is complicated. I have no idea how to solve it. Can I develop a software that can help me execute this? it's as simple as [switchhost](https://github.com/oldj/SwitchHosts) used to switch proxy host.

**Comment by ademakdogan** [comment, marked as answer]
Yes, it is possible to develop a software or tool that can help you manage your multiple GitHub and GitLab accounts. One approach could be to use a command-line tool like Git, and then configure it to use different SSH keys and configurations for each of your accounts. This way, you can switch between your accounts by changing the configuration.

Here's an example of how you can set up Git to use multiple SSH keys:

1- Generate a new SSH key for each of your accounts:

`ssh-keygen -t ed25519 -C "your_email@example.com" -f ~/.ssh/github_account1`
`ssh-keygen -t ed25519 -C "your_email@example.com" -f ~/.ssh/gitlab_account1`

2- Add the public keys (*.pub files) to your GitHub and GitLab accounts.

3- Create a new SSH configuration file for each account:

Bash
`touch ~/.ssh/config.github_account1`
`touch ~/.ssh/config.gitlab_account1`
4- Add the following lines to each configuration file:

Host github.com-account1
.....HostName github.com
.....User git
.....IdentityFile ~/.ssh/github_account1

Host gitlab.com-account1
.....HostName gitlab.com
.....User git
.....IdentityFile ~/.ssh/gitlab_account1

5- Replace account1 with your desired account name, and repeat steps 1-4 for each of your accounts.

6- Now, when you clone a repository from GitHub or GitLab, use the github.com-account1 or gitlab.com-account1 hostname instead of the default hostname. For example:

`git clone git@github.com-account1:username/repo.git`

**Reply by vaynevayne** [reply]
It is well written, as the saying goes,
>  It's easy to someone who is able to do it;but the contary to those who aren't.

 In addition to the second step, other steps will be completed through software. I believe many people have this requirement.
